packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -149,16 +149,33 @@ Future<void> main() async {
 }
 ''';
     } else {
+
+// TODO: Add the FlutterLoader code to the if-case above (with plugins)!
+
       contents = '''
 // @dart=${languageVersion.major}.${languageVersion.minor}
 
 import 'dart:ui' as ui;
 
+import 'package:flutter_web_plugins/flutter_loader.dart';
+
 import '$mainImport' as entrypoint;
 
 Future<void> main() async {
-  await ui.webOnlyInitializePlatform();
-  entrypoint.main();
+  if (isFlutterJsLoaderConfigured) {
+    FlutterLoader(
+      initEngine: () async {
+        await ui.webOnlyInitializePlatform();
+      },
+      runApp: () {
+        entrypoint.main();
+      }
+    )..notifyFlutterReady();
+  } else {
+    // Legacy mode
+    await ui.webOnlyInitializePlatform();
+    entrypoint.main();
```

Comment:
I suggest we try to figure out a way to support legacy mode in `FlutterLoader` instead of keeping this if-else everywhere. Maybe a subclass `LegacyFlutterLoader`?

Reply:
I agree. We have these situations on the engine side w.r.t. cross-browser support, and abstraction has done a better job of keeping the code clean compared to if/else blocks.

Reply:
Yes, this will happen before the release; the Loader will be initialized regardless, but it'll have 2 modes of operation, depending on it finding the new configuration or not.

Reply:
This is now done. webOnlyWarmupEngine will know in which mode it is, and either return an object that can initialize the code through JS, OR directly initialize the app as it was before. This much simplifies the _wrapper_ code that needs to be generated. (This has been implemented in the new PRs)